src/features/cluster/delete-dialog/main/clear-as-deleteing-channel-listener.injectable.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getRequestChannelListenerInjectable } from "../../../../common/utils/channel/request-channel-listener-injection-token";
+import { clearClusterAsDeletingChannel } from "../common/clear-as-deleting-channel";
+import clearClusterAsDeletingHandlerInjectable from "./clear-as-deleting-handler.injectable";
+
+const clearClusterAsDeletingChannelListenerInjectable = getRequestChannelListenerInjectable({
```

Comment:
Hmm. The previous solution needed only one injectable to listen and act for a channel, and the new solution requires two, ie., a "listener" and a "handler"? Thinking real hard if this can bite us. We could put some make-up on this with that injectable-sibling thing in the horizon?

Reply:
Isn't type safety worse if it relies on usage of `getRequestChannelListenerInjectable`, because injectables created outside of it can be unsafe?

Reply:
I'm not sure if ability to "inject the handler separately" is aspirational, as it messages multiple responsibilities (ie. not SRP). However, if the handler does something that other places might be interested in, we can extract that to be injected by the other places. They probably didn't need a "handler" anyway, but instead something more specific.